cli/src/cli_util.rs

```diff
@@ -2194,14 +2194,17 @@ pub fn write_config_value_to_file(
                 ))
             })?;
     }
-    // Error out if overwriting non-scalar value for key (table or array).
+    // Error out if overwriting non-scalar value for key (table or array) with
+    // scalar.
     match target_table.get(last_key_part) {
         None | Some(toml_edit::Item::None) => {}
         Some(toml_edit::Item::Value(val)) if !val.is_array() && !val.is_inline_table() => {}
         _ => {
-            return Err(user_error(format!(
-                "Failed to set {key}: would overwrite entire non-scalar value with scalar"
-            )));
+            if !item.is_array() && !item.is_inline_table() {
+                return Err(user_error(format!(
+                    "Failed to set {key}: would overwrite entire non-scalar value with scalar"
+                )));
+            }
```

Comment:
Perhaps, we can simply check if the existing item is structurally non-value table/array?

```rust
// or use `.is_xxx()`
None | Some(toml_edit::Item::None | toml_edit::Item::Value(_)) => {}
Some(toml_edit::Item::Table(_) | toml_edit::Item::ArrayOfTables(_)) => return Err(..),
```

I think it's okay to override `colors.x = { bold = true }` with scalar value. It's just bad if we allowed to override the entire `colors` table.

Reply:
That makes sense. Thanks for the suggestion. PTAL at the updated version. I changed the error message too. I made it say "would overwrite entire table" without mentioning arrays of tables to keep it simple.